test/test-unminify.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const mocha = require('mocha');
+const unminify = require('../utils/unminify');
+const sourceMap = require('source-map');
+const chai = require('chai');
+const sinon = require('sinon');
+const Request = require('../utils/request');
+const describe = mocha.describe;
+const it = mocha.it;
+const before = mocha.before;
+const after = mocha.after;
+const expect = chai.expect;
+
+
+describe('Test unminification', function() {
+  const rawSourceMap = {
+    version: 3,
+    file: 'min.js',
+    names: ['bar', 'baz', 'n'],
+    sources: ['one.js', 'two.js'],
+    sourceRoot: 'http://example.com/www/js/',
+    mappings: 'CAAC,IAAI,IAAM,SAAUA,GAClB,' +
+       'OAAOC,IAAID;CCDb,IAAI,IAAM,SAAUE,GAClB,OAAOA',
+  };
+  const sandbox = sinon.sandbox.create();
+  before(function() {
+    sandbox.stub(Request, 'request').callsFake(function(url, callback) {
+      setTimeout(function() {
+        callback(null, null, JSON.stringify(rawSourceMap));
+      }, 10);
+    });
+  });
+
+  after(function(done) {
+    sandbox.restore();
+    done();
+  });
+
+  it('Should unminify a stack trace line given a source map', function() {
+    const sourceMapConsumer = new sourceMap.SourceMapConsumer(rawSourceMap);
+    const location = {
+          lineNumber: 2,
+          columnNumber: 28,
+          sourceUrl: 'https://example.com/www/js/min.js',
+          stackTraceLine: ' at https://example.com/www/js/min.js:2:28',
+        };
+    expect(unminify.unminifyLine(location, sourceMapConsumer)).
+        to.equal(' at http://example.com/www/js/two.js:2:10');
+  });
+
+  it('Should make only one network request per source map', function() {
+    const stackTrace = ['http://example.com/www/js/two.js.map',
+      'http://example.com/www/js/two.js.map'];
+    const promises = unminify.extractSourceMaps(stackTrace);
+    expect(promises[0] === promises[1]).to.equal(true);
+  });
+
+  it('Should use a source map from cache if already cached', function() {
```

Comment:
This test doesn't test exactly this, and essentially boils down to the previous test. We need to test specifically for concurrent calls (not the same call):

1. [ two.js, one.js ]
2. [ two.js, two.js ]